### PR TITLE
Support PHP 8.2, Remove PHP 7.4

### PR DESCRIPTION
**Pull request by gsteel**
|    Q          |   A
|-------------- | ------
| New Feature   | yes
| RFC           | yes

### Description

Made a start on this, but there is *so much* mocking, I'm losing the will to live.

If preferred, I can revert the 8.2 support and merge as-is… It's gonna take ages to get rid of prophecy here…

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -36,6 +36,7 @@
     },
     "require": {
         "php": "~8.0.0 || ~8.1.0 || ~8.2.0",
+        "lcobucci/jwt": "^4",
```
Comment (Ocramius) [inline composer.json]: Wait, this wasn't a dependency before? 🤔

**Comment by gsteel** [comment]
`lcobucci/jwt` needed bumping to ^4 for 8.2 lowest to pass, originally a transitive dep of `oauth2-server`. Should this be moved to a dev-dep instead @Ocramius?

**Comment by Ocramius** [comment]
Imo we could add a conflict with earlier versions, rather than introducing a hard dependency here

**Comment by gsteel** [comment]
Thanks @Ocramius :)

**Review by Ocramius — APPROVED** [review]
Huge baseline reduction: well done, @gsteel!

**Pull request merged**